blog/content/post/2025-05-07-dasharo-ci.md

```diff
@@ -0,0 +1,337 @@
+      uses: actions/checkout@v4
+      with:
+        path: edk2
+
+    - name: Checkout microsoft/secureboot_objects
+      uses: actions/checkout@v4
+      with:
+        repository: microsoft/secureboot_objects
+        path: secureboot_objects
+
+    - name: Update DBX blob
+      run: |
+        cp secureboot_objects/PostSignedObjects/DBX/amd64/DBXUpdate.bin edk2/DasharoPayloadPkg/SecureBootDefaultKeys/DBXUpdate.bin
+
+    - name: Set current date
+      run: |
+        pushd secureboot_objects
+        echo "RELEASE_DATE=$(git log -1 --pretty='format:%cs' PostSignedObjects/DBX/amd64/DBXUpdate.bin)" >> ${GITHUB_ENV}
+        popd
+
+    - name: Submit pull request
+      uses: peter-evans/create-pull-request@v7.0.7
+      with:
+        path: edk2
+        base: dasharo
+        branch: update_dbx_${{ env.RELEASE_DATE }}
+        title: Update DBX ${{ env.RELEASE_DATE }}
+        commit-message: "[automated change] Update DBX ${{ env.RELEASE_DATE }}"
+```
+
+We see the same overall logic:
+
+- Clone the secureboot objects repo
+- Make a checksum of the current dbx
+- Make a checksum of the latest dbx
+- Compare them
+- If the sums don't match:
+  - update the file
+  - create PR
+
+That's pretty much it. These two GitHub workflows automate updates of both
+microcode and the revocation database.
+
+## Closing thoughts
+
+Introducing these automatic checks makes our firmware not only more secure,
+but also more transparent. As the repositories and their CI workflows are open,
+each user can see for themselves when the microcode and DBX were last updated,
+and build from the main branch themselves with a guarantee that these components
+will be up to date each time.
+
+Other BIOS firmware vendors typically don't provide this information, and if
+they do, it's buried in the release notes. It's often not clear if the microcode
+is indeed the latest version available from the CPU vendor. Meanwhile, Dasharo
+release notes [contain detailed SBoM](https://docs.dasharo.com/variants/novacustom_v540tu/releases_heads/#v090-2025-03-20)
+(Software Bill of Materials) sections describing exactly what microcode you're
+getting:
+
+![V540TU Heads v0.9.0 SBoM](/img/v540tu_sbom.png)
+
+We hope the
+introduction of these checks will make our firmware safer and more worthy of
+our users' trust.
+
+Unlock the full potential of your hardware and secure your firmware with the
+experts at 3mdeb! If you're looking to boost your product's performance and
+protect it from potential security threats, our team is here to help.
+[Schedule a call with us](https://calendly.com/3mdeb/consulting-remote-meeting)
+or drop us an email at `contact<at>3mdeb<dot>com` to start unlocking the hidden
+benefits of your hardware. And if you want to stay up-to-date on all things
+firmware security and optimization, be sure to
+[sign up for our newsletter](https://3mdeb.com/subscribe/3mdeb_newsletter.html).
+Don't let your hardware hold you back, work with 3mdeb to achieve more!
```

Comment:
@mkopec I would appreciate CfP https://cfp.3mdeb.com/developers-vpub-0xf-2025/ submission as well as promotion of the event: https://events.dasharo.com/event/5/dasharo-user-group-dug-10-and-dasharo-developers-vpub-0xf

You can check recently published posts to see how it was done for the Zarhus Developers Meetup. Also, please give me a clear indicator when you finish improvements, since I'm not sure if I should wait for something more, e.g., regarding [this comment](https://github.com/3mdeb/news-and-ideas/pull/659#pullrequestreview-2834221524)

Reply:
> added link to DUG here: [b68e172](https://github.com/3mdeb/news-and-ideas/commit/b68e1720dd7734f6d828aea268f6b74424b8445c)

Which line? Sorry, can't see it.

> 
> > I would appreciate CfP
> 
> Are you asking me to put in a link to the CfP, or to submit my talk to about this topic to the CfP? :) I plan on submitting a CfP once this blogpost lands

Ok, blog post is almost finished. Please submit the CfP, that is what I mean.

Reply:
CfP submitted